shenanigan/models/inception/run.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+import tensorflow as tf
+
+from shenanigan.models.inception.model import build
+from shenanigan.utils.data_helpers import get_record_paths
+from shenanigan.utils.utils import mkdir
+
+
+def _parse_function(proto, classes):
+    f = {
+        'image_small': tf.io.FixedLenFeature([], tf.string),
+        'image_large': tf.io.FixedLenFeature([], tf.string),
+        'wrong_image_small': tf.io.FixedLenFeature([], tf.string),
+        'wrong_image_large': tf.io.FixedLenFeature([], tf.string),
+        'name': tf.io.FixedLenFeature([], tf.string),
+        'text': tf.io.FixedLenFeature([], tf.string),
+        'label': tf.io.FixedLenFeature([], tf.int64),
+    }
+    parsed_features = tf.io.parse_single_example(proto, f)
+    img = tf.io.decode_image(parsed_features['image_large'], dtype=tf.float32)
+    label = tf.one_hot(parsed_features['label'], depth=classes)
+    return img, label
+
+def load_dataset(input_path, batch_size, shuffle_buffer, classes):
+    dataset = tf.data.TFRecordDataset(input_path)
+    dataset = dataset.map(lambda x: _parse_function(x, classes), num_parallel_calls=16)
+    dataset = dataset.shuffle(shuffle_buffer).batch(batch_size).prefetch(1)
```

Comment:
Only pre-fetch 1? Might this be faster if that is increased?

Reply:
Yeah I think it will, except it crashed my machine haha accidentally had batch size set to 640 so that was the cuprite. I saw somewhere recommend prefetching "batch size" but think that will kill the RAM. Will increase it to like 2-5 for now.